### PR TITLE
[Authz] Implicitly extracting name of Extractors and OperationType in spec file generation preventing use class name String directly

### PR DESCRIPTION
**Pull request by bowenliang123**
### _Why are the changes needed?_

- add implicit methods `classSimpleName` and `operationTypeStr` to package object `org.apache.kyuubi.plugin.spark.authz` in test sources
- increase compilation alignment to extractor itself, and easier use for future third party adaptation avoiding calling wrong `getName` of the calss
- preventing use String or class name as magic value directly which easily causes mistakes
- no changes with spec JSON file regeneration after this PR 

### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by bowenliang123** [comment]
cc @yaooqinn

**Comment by yaooqinn** [comment]
The extractor name has not necessarily to be simple classname

**Comment by bowenliang123** [comment]
> The extractor name has not necessarily to be simple classname

Or let's change the method name to `classSimpleName` in AuthzUtils ?

**Comment by yaooqinn** [comment]
Let's define an `implicit def` in the test package instead of `AuthZUtils`. Maybe a package object in `o.a.k.p.s.a.gen` or `JsonSpecFileGenerator`  

```scala  
implicit def classSimpleName[T](clz: Class[_]): String = clz.getSimpleName
```

And then, we can use it on the caller side like 
```scala
DatabaseDesc("databaseName", classOf[StringDatabaseExtractor])
```

Similarly, we can also have an `implicit def` for OperationType

```scala
implicit def operationTypeStr(t: OperationType): String = t.toString
```

File: extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/gen/package.scala [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.plugin.spark.authz
+
+import scala.reflect.ClassTag
+
+package object gen {
+  implicit def classSimpleName[T](implicit ct: ClassTag[T]): String = {
```
Comment (yaooqinn) [inline extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/gen/package.scala:23]: If you define it in such a way and call it explicitly, the `implicit` here is not necessary

Comment (bowenliang123) [reply extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/gen/package.scala]: OK, let's use it in an implicit way as you suggested example above.

**Review by yaooqinn — APPROVED** [review]
LGTM

**Comment by yaooqinn** [comment]
thanks. merged to master

**Pull request closed without merging**